### DISCUSSION TITLE
Debian / Ubuntu Packages

### DISCUSSION THREAD

**Discussion by Kr0san89** [valkey-io/valkey#629] [General]
Hello,

are there any plans du provide packages for debian / ubuntu to get via apt?

thanks

**Comment by tibz7** [comment]
same question here

**Comment by njbhatt18** [comment]
Currently, no Ubuntu package is available. For testing purposes, I compiled from the source.

**Comment by madolson** [comment]
There are versions available from percona here: https://www.percona.com/blog/new-valkey-packages-by-percona/. 

It's not yet available through other official distribution channels, but it'll get there soon enough.

**Reply by melroy89** [reply]
For now I advise people to download the latest binary artifact instead (v8.0.1 at this moment) from: https://valkey.io/download/releases/v8-0-1/ if you don't want to setup / use Percona.

Untar. And copy the directory structure + files into `/usr/local`  yourself.

Binaries have TLS and systemd support enabled.

**Reply by njbhatt18** [reply]
> There are versions available from percona here: https://www.percona.com/blog/new-valkey-packages-by-percona/.
> 
> It's yet to be available through other official distribution channels, but it'll get there soon enough.

Percona apt repo is still not production ready; I tried to install it in Ubuntu 22, but I got an error to start service even though uninstalling the command was not working, and I could not uninstall it. I need to update the config file, stop services manually and then uninstall work.

**Comment by njbhatt18** [comment]
Is there any plan to provide binary for the latest Ubuntu versions, like 22 or 24?

**Reply by melroy89** [reply]
Just grap the lastest build artifact like for Ubuntu 22.04 focal from here https://valkey.io/download/releases/v8-0-1/. This also works for Ubuntu 24.04

Direct link would be: https://download.valkey.io/releases/valkey-8.0.1-focal-x86_64.tar.gz

**Comment by binaryfire** [comment]
FYI Valkey is now included in Ubuntu 24.04 LTS and 24.10: https://ubuntu.com/blog/valkey-ubuntu

**Reply by melroy89** [reply]
Valkey 7* not 8

**Reply by binaryfire** [reply]
Ah right. Thanks.

**Reply by melroy89** [reply]
Ubuntu 25.04 will have Valkey 8.

**Comment by lvoytek** [comment]
Valkey 8 is now available in Debian Unstable and Testing. It will also be available in the [25.04 release of Ubuntu](https://launchpad.net/ubuntu/+source/valkey)

**Comment by hubeni** [comment]
Great job so far :)
I was wondering why Valkey 7.2.7 is not available for Ubuntu 24.04 as from my understanding it should be non breaking and provides security patches? 
Are there plans to keep Ubuntu 24.04 also up to date?

**Reply by lvoytek** [reply]
Thanks for the ping :)
I am working on getting 7.2.7 into 24.04 and 24.10 now. I've created a bug on Launchpad to provide updates on the process [here](https://bugs.launchpad.net/ubuntu/+source/valkey/+bug/2091129).

**Reply by hubeni** [reply]
@lvoytek Are there plans to upload all releases as Ubuntu packages? 
I've seen there's a new release of 7.2.8 fixing some important security vulnerabilities which is not available yet.
Is that something that can be automated or is it always manual effort? :)

**Reply by lvoytek** [reply]
@hubeni I plan on updating active Ubuntu releases with valkey security fixes as quickly as possible. Unfortunately it takes manual effort when doing a full version update to make sure no backwards-incompatible changes are introduced to our users. I'm working with the Stable Release Updates team to speed things up over time though

**Reply by melroy89** [reply]
An Ubuntu PPA on launchpad.net could also help maybe? Just thinking out loud here.

**Reply by lvoytek** [reply]
For what it's worth I work on the updates using an unofficial PPA [here](https://launchpad.net/~lvoytek/+archive/ubuntu/valkey-sru). It currently has 7.2.8 working in 24.04 and 24.10. It would be ideal to have something more official though

**Reply by hubeni** [reply]
Will version 7.2.9 be also provided as Ubuntu package?
Thanks in advance.

**Reply by lvoytek** [reply]
Thanks for bringing this up! I created a bug [here](https://bugs.launchpad.net/ubuntu/+source/valkey/+bug/2115258) to track progress.

**Reply by hubeni** [reply]
@Ivoytek Any updates on that ? 
Would be great to have v. 7.2.10 for Noble.
Thanks in advance.

**Reply by lvoytek** [reply]
@hubeni currently 7.2.10 is in the queue awaiting approval for inclusion in Noble - https://launchpad.net/ubuntu/noble/+queue?queue_state=1&queue_text=valkey
Same for 8.0.4 in Plucky
It should be released soon :)

**Reply by hubeni** [reply]
@Ivoytek is it common that those approvals take more than a month as it is still not approved?

**Reply by lvoytek** [reply]
@hubeni Sorry for the wait, these reviews usually happen a lot faster but they can vary. 7.2.10 is available in noble-updates now though

**Reply by hubeni** [reply]
@Ivoytek
Coud you please provide also 7.2.11 to the Ubuntu Package Manager as it includes fixes for critical security vulnerabilities?
And is there a way to automate the provisioning of any new Valkey release to the Ubuntu Package Manager? :-)

**Reply by lvoytek** [reply]
@hubeni Here is the current bug report for valkey updates for reference https://bugs.launchpad.net/ubuntu/+source/valkey/+bug/2127122
Unfortunately updates to Ubuntu stable releases cannot be automated as they require an extensive review process to avoid breaking users' systems. For more info see [Ubuntu's SRU documentation](https://documentation.ubuntu.com/project/how-ubuntu-is-made/processes/stable-release-updates/). To speed things up though we should soon have an [SRU exception](https://documentation.ubuntu.com/sru/en/latest/reference/package-specific/) for the package which will make backporting new versions much faster.

**Comment by orkutmuratyilmaz** [comment]
The tutorial here is misleading: https://valkey.io/topics/installation/

**Comment by bilkoua** [comment]
Hi everyone!
In my opinion, a full-fledged official repository, similar to `packages.redis.io`, is absolutely necessary.
Unfortunately, the **official Debian repository** contains very outdated packages that are only available via backports and doesn’t include version 7. Version 7 could still be useful for some users, especially if their clients haven’t yet adopted Valkey 8.
Regarding **Percona’s packages**, they are also outdated and of poor quality.
As a temporary solution, we are currently building the packages ourselves and pushing them to our private corporate repository. However, having an official repository with up-to-date versions would be a much better option.
I’m ready to invest my time to help make this happen, but we need a dedicated repository for it. The good news is that this doesn’t require significant financial resources.

There are free options available:

- **GitHub Pages** (like this - [link](https://gist.github.com/ikbelkirasan/57e60bbcaecd0bd5cf2cbf441e744159) + custom domain)
- **Packagecloud.io** (free for open-source projects - [link](https://blog.packagecloud.io/packagecloud-partners-with-stackstorm-to-support-the-oss-community/))
- **Cloudsmith.com** (offers 50GB of storage and 200GB of monthly bandwidth for free - [link](https://cloudsmith.com/blog/cloudsmith-loves-opensource))

Alternatively, you can use the existing infrastructure:
- You likely already have an S3 bucket in use for https://download.valkey.io/ ([example](https://github.com/valkey-io/valkey-release-automation/blob/main/.github/workflows/call-build-linux-x86-packages.yml#L87)).

I’m happy to assist with this - let me know your thoughts!

**Comment by pitch-d** [comment]
why haven't valkey managed to create a stable package repository for debian/redhat
i know that jordan sissel started fpm +10 years ago to sort this out but today it should be sorted out or?

**Comment by bilkoua** [comment]
Unfortunately, I kept waiting for a proper repo with separated versions and up-to-date releases, but it never happened, so I built one myself.
I set up a GitHub Actions CI pipeline that checks for new upstream versions, and when it finds one, it builds the package and publishes it as an APT repository format on **Cloudflare R2**.
Currently, the following targets are supported:
| Distribution | Codename | Architecture |
| ------------ | -------- | ------------ |
| Debian 13 | `trixie` | `amd64`, `arm64` |
| Debian 12 | `bookworm` | `amd64`, `arm64` |
| Ubuntu 24.04 LTS | `noble` | `amd64`, `arm64` |
| Ubuntu 22.04 LTS | `jammy` | `amd64`, `arm64` |

Adding more is straightforward if needed.
What I need now is help with testing.

Valkey repo page: https://pkgs.bil.co.ua/pages/valkey.html
Git repo: [https://github.com/community-pkgs/packages/tree/main/packages/valkey](https://github.com/community-pkgs/packages/tree/main/packages/valkey)
